### PR TITLE
migrate dagster-airflow to crag

### PR DESCRIPTION
**Pull request by yuhan**
## Summary

add `job_name` to existing apis and deprecate `pipeline_name` there:
- `make_airflow_dag`
- `make_airflow_dag_for_operator`
- `make_airflow_dag_containerized`
- `dagster-airflow scaffold` cli

introduce new api: 
- `make_dagster_job_from_airflow_dag`

## Test Plan
bk

## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: python_modules/libraries/dagster-airflow/dagster_airflow/dagster_job_factory.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    Returns:
+        job_def (PipelineDefinition): The generated Dagster job
+
+    """
+    pipeline_def = make_dagster_pipeline_from_airflow_dag(
+        dag, tags, use_airflow_template_context, unique_id
+    )
+    # pass in tags manually because pipeline_def.graph doesn't have it threaded
+    return pipeline_def.graph.to_job(tags={**pipeline_def.tags})
```
[the remark this answers is not in the document]
Comment (sryza) [reply python_modules/libraries/dagster-airflow/dagster_airflow/dagster_job_factory.py:57]: this is a little brittle (e.g. what if we added resources or something), but should be fine

File: python_modules/libraries/dagster-airflow/dagster_airflow/dagster_job_factory.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            `tags={'airflow_execution_date': utc_date_string}` to specify execution_date used within
+            execution of Airflow Operators.
+        use_airflow_template_context (bool): If True, will call get_template_context() on the
+            Airflow TaskInstance model which requires and modifies the DagRun table.
+            (default: False)
+        unique_id (int): If not None, this id will be postpended to generated op names. Used by
+            framework authors to enforce unique op names within a repo.
+
+    Returns:
+        job_def (JobDefinition): The generated Dagster job
```
Comment (sryza) [inline python_modules/libraries/dagster-airflow/dagster_airflow/dagster_job_factory.py:50]: No need to name return values - I'd be surprised if sphinx supported it.

File: python_modules/libraries/dagster-airflow/dagster_airflow/factory.py [hunk 3]
```diff
@@ -240,14 +241,15 @@ def _make_airflow_dag(
...
     run_config=None,
     mode=None,
     instance=None,
     dag_id=None,
     dag_description=None,
     dag_kwargs=None,
     op_kwargs=None,
+    pipeline_name=None,
 ):
     """Construct an Airflow DAG corresponding to a given Dagster pipeline.
```
Comment (sryza) [inline python_modules/libraries/dagster-airflow/dagster_airflow/factory.py:254]: change this to job?

**Review by sryza — COMMENTED** [review]
This approach looks good to me

File: python_modules/libraries/dagster-airflow/dagster_airflow/factory.py (hunk 3 above)
Comment (yuhan) [reply python_modules/libraries/dagster-airflow/dagster_airflow/factory.py:254]: did you mean changing "job_name" to "job"

Comment (sryza) [reply python_modules/libraries/dagster-airflow/dagster_airflow/factory.py:254]: I mean "Construct an Airflow DAG corresponding to a given Dagster job"

Comment (yuhan) [reply python_modules/libraries/dagster-airflow/dagster_airflow/factory.py:254]: ah misread the github highlighting.. roger

File: python_modules/libraries/dagster-airflow/dagster_airflow/cli.py [hunk 4]
```diff
@@ -105,7 +104,7 @@ def main():
 @click.option(
     "--module-name", "-m", type=click.STRING, help="The name of the source module", required=True
 )
-@click.option("--pipeline-name", type=click.STRING, help="The name of the pipeline", required=True)
+@click.option("--pipeline-name", type=click.STRING, help="The name of the name")
```
Comment (sryza) [inline python_modules/libraries/dagster-airflow/dagster_airflow/cli.py:107]: This should be the "The name of the pipeline", not "The name of the name", right?

**Review by sryza — APPROVED** [review]
Looks great!

File: python_modules/libraries/dagster-airflow/dagster_airflow/cli.py (hunk 4 above)
Comment (yuhan) [reply python_modules/libraries/dagster-airflow/dagster_airflow/cli.py]: err thanks for catching!

**Pull request merged**